libraries/botbuilder/src/skills/skillHttpClient.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Activity, ConversationReference, TurnContext } from 'botbuilder-core';
+import { ICredentialProvider } from 'botframework-connector';
+import { InvokeResponse } from '../botFrameworkAdapter';
+import { BotFrameworkHttpClient } from '../botFrameworkHttpClient';
+import { BotFrameworkSkill } from './botFrameworkSkill';
+import { SkillConversationIdFactoryBase } from './skillConversationIdFactoryBase';    
+
+/**
+ * A BotFrameworkHttpClient specialized for Skills that encapsulates Conversation ID generation.
+ */
+export class SkillHttpClient extends BotFrameworkHttpClient {
+    constructor(credentialProvider: ICredentialProvider, private readonly conversationIdFactory: SkillConversationIdFactoryBase, channelService?: string) {
+        super(credentialProvider, channelService);
+        if (!this.conversationIdFactory) {
+            throw new Error('conversationIdFactory missing');
+        }
+    }
+
+    public async postToSkill(fromBotId: string, toSkill: BotFrameworkSkill, serviceUrl: string, activity: Activity): Promise<InvokeResponse> {
```

Comment:
>postToSkill [](start = 17, length = 11)

In C#, this method is called PostActivity instead of PostToSkill, consider matching the name.

Reply:
That's not possible in TypeScript 😞